packages/util/src/logging.ts

```diff
@@ -0,0 +1,36 @@
+import { Logger } from 'ts-log';
+
+export const loggerMethodNames: (keyof Logger)[] = ['debug', 'error', 'fatal', 'info', 'trace', 'warn'];
+
+/**
+ * Appends an object containing context metadata to each log entry
+ * Nested contexts can be added by deriving multiple times.
+ *
+ * Example: single context
+ *   - const oneContextLogger = contextLogger('ctx1', rawLogger);
+ *   - oneContextLogger.debug('hi') calls `rawLogger({context: ['ctx1']}, hi)
+ *
+ * Example: nested context
+ *   - const twoContextLogger = contextLogger('ctx2', oneContextLogger);
+ *   - twoContextLogger.debug('hi') calls `rawLogger({context: ['ctx1', 'ctx2']}, hi)
+ *
+ * Example: nested context with merged message object
+ *   - const mergeLogger = contextLogger('ctx2', oneContextLogger);
+ *   - mergeLogger.debug({info: 'Superb'}, 'hi') calls `rawLogger({context: ['ctx1', 'ctx2'], moreInfo: 'Superb'}, hi)
+ */
+export const contextLogger = (logger: Logger, context: string): Logger => ({
+  context: [...(logger.context || []), context],
+  ...(<Logger>(<unknown>Object.fromEntries(
+    loggerMethodNames.map<[keyof Logger, Logger['info']]>((method) => [
+      method,
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      (message?: any, ...optionalParams: any[]) => {
+        const contextArg = { context: [...(logger.context || []), context] };
```

Comment:
I think you're right about the formatted form.
I am trying to figure out how it will work with [bunyan APi](https://github.com/trentm/node-bunyan#log-method-api), which allows providing a "fields" object as first arg, which is merged into the log record (e.g. see Error special case).
If we add a formatted context string as first arg, we shadow this functionality to users wanting to derive from `bunyan`